tests/topotests/bgp-evpn-vxlan_topo1/test_bgp_evpn_vxlan.py

```diff
@@ -0,0 +1,275 @@
+# for any purpose with or without fee is hereby granted, provided
+# that the above copyright notice and this permission notice appear
+# in all copies.
+#
+# THE SOFTWARE IS PROVIDED "AS IS" AND NETDEF DISCLAIMS ALL WARRANTIES
+# WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+# MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL NETDEF BE LIABLE FOR
+# ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY
+# DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS,
+# WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS
+# ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE
+# OF THIS SOFTWARE.
+#
+
+"""
+test_bgp_evpn_vxlan.py: Test VXLAN EVPN MAC a route signalling over BGP.
+"""
+
+import os
+import sys
+import json
+from functools import partial
+import pytest
+
+# Save the Current Working Directory to find configuration files.
+CWD = os.path.dirname(os.path.realpath(__file__))
+sys.path.append(os.path.join(CWD, '../'))
+
+# pylint: disable=C0413
+# Import topogen and topotest helpers
+from lib import topotest
+from lib.topogen import Topogen, TopoRouter, get_topogen
+from lib.topolog import logger
+
+# Required to instantiate the topology builder class.
+from mininet.topo import Topo
+
+class TemplateTopo(Topo):
+    "Test topology builder"
+    def build(self, *_args, **_opts):
+        "Build function"
+        tgen = get_topogen(self)
+
+        # This function only purpose is to define allocation and relationship
+        # between routers, switches and hosts.
+        #
+        #
+        # Create routers
+        tgen.add_router('P1')
+        tgen.add_router('PE1')
+        tgen.add_router('PE2')
+        tgen.add_router('host1')
+        tgen.add_router('host2')
+
+
+        # Host1-PE1
+        switch = tgen.add_switch('s1')
+        switch.add_link(tgen.gears['host1'])
+        switch.add_link(tgen.gears['PE1'])
+
+        # PE1-P1
+        switch = tgen.add_switch('s2')
+        switch.add_link(tgen.gears['PE1'])
+        switch.add_link(tgen.gears['P1'])
+
+        # P1-PE2
+        switch = tgen.add_switch('s3')
+        switch.add_link(tgen.gears['P1'])
+        switch.add_link(tgen.gears['PE2'])
+
+        # PE2-host2
+        switch = tgen.add_switch('s4')
+        switch.add_link(tgen.gears['PE2'])
+        switch.add_link(tgen.gears['host2'])
+
+def setup_module(mod):
+    "Sets up the pytest environment"
+    # This function initiates the topology build with Topogen...
+    tgen = Topogen(TemplateTopo, mod.__name__)
+    # ... and here it calls Mininet initialization functions.
+    tgen.start_topology()
+
+    pe1 = tgen.gears['PE1']
+    pe2 = tgen.gears['PE2']
+    p1 = tgen.gears['P1']
+    
+    # set up PE bridges with the EVPN member interfaces facing the CE hosts
+    pe1.run('ip link add name br101 type bridge stp_state 0')
+    pe1.run('ip link set dev br101 up')
+    pe1.run('ip link add vxlan101 type vxlan id 101 dstport 4789 local 1.1.1.1 nolearning')
```

Comment:
could you replace public routable IPs to private subnets or documentation subnets?